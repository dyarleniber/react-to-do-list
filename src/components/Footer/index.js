import React from 'react';

const Footer = ({ length }) => {
    return (
        <small> TOTAL: {length} </small>
    );
};

export default Footer;
