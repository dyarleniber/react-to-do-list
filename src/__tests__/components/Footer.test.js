import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import Footer from '../../components/Footer';

const middlewares = [];
const mockStore = configureStore(middlewares);

describe('Footer component', () => {
  it('should render without crashing', () => {
    let wrapper;
    let initialState;
    let store;

    initialState = { todos: [] };
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <Footer />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();

    initialState = {
      todos: [
        {
          id: 1,
          text: 'Todo 1',
          completed: true,
        },
        {
          id: 2,
          text: 'Todo 2',
          completed: false,
        },
      ],
    };
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <Footer />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render total', () => {
    let wrapper;
    let length;
    let total;
    let initialState;
    let store;

    length = 2;
    total = <small>{`TOTAL: ${length}`}</small>;
    initialState = {
      todos: [
        {
          id: 1,
          text: 'Todo 1',
          completed: true,
        },
        {
          id: 2,
          text: 'Todo 2',
          completed: false,
        },
      ],
    };
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <Footer />
      </Provider>,
    );
    expect(wrapper.contains(total)).toEqual(true);

    length = 0;
    total = <small>{`TOTAL: ${length}`}</small>;
    initialState = { todos: [] };
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <Footer />
      </Provider>,
    );
    expect(wrapper.contains(total)).toEqual(false);
  });
});
