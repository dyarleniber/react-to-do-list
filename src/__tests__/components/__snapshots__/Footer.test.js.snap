// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should render without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Footer>
    <footer
      className="text-center"
    />
  </Footer>
</Provider>
`;

exports[`Footer component should render without crashing 2`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Footer>
    <footer
      className="text-center"
    >
      <small>
        TOTAL: 2
      </small>
    </footer>
  </Footer>
</Provider>
`;
